src/schema-association-service.ts

```diff
@@ -243,6 +267,33 @@ async function autoDetectSchema(
     // Create the global storage folder to guarantee that it exists.
     await vscode.workspace.fs.createDirectory(context.globalStorageUri);
 
+    // Try to fetch schema in the following order:
+    // 1. Cached 1ESPT schema
+    // 2. 1ESPT schema if user is signed in with microsoft account and has enabled 1ESPT schema
+    // 3. Cached Organization specific schema
+    // 4. Organization specific schema
+
+    // get cached 1ESPT schema if:
+    // 1. User is signed in with microsoft account
+    // 2. 1ESPT schema is enabled
+    // 3. 1ESPT schema is not older than 24 hours
+    const [schemaUri1ESPT , skipOrgSpecificCachedSchema] = getCached1ESPTSchemaInformation(context, organizationName, session, oneesptSchemaEnabled, lastUpdated1ESPTSchema, seen1ESPTOrganizations);
+    if(schemaUri1ESPT){
+        return schemaUri1ESPT;
+    }
+
+    const token = await session.credentials2.getToken();
+    const authHandler = azdev.getBearerHandler(token.accessToken);
+    const azureDevOpsClient = new azdev.WebApi(`https://dev.azure.com/${organizationName}`, authHandler);
+
+    // if user is signed in with microsoft account and has enabled 1ESPipeline Template Schema, then give preference to 1ESPT schema
+    if(oneesptSchemaEnabled){
+        const schemaUri1ESPT = await get1ESPTSchemaUriIfAvailable(azureDevOpsClient, organizationName, session, context, lastUpdated1ESPTSchema, seen1ESPTOrganizations);
```

Comment:
lastUpdated1ESPTSchema should be per-org, no? Also, does the 1ES PT schema change so often that it needs to be refreshed daily instead of on-open like the existing schema?

Reply:
Modified lastUpdated1ESPTSchema so that it is org wise.

 1ESPTSchema gets rolled out every week. We have multiple rings for rollout. So every org will get the schema on a different day of the week. Refreshing it every day makes sure that we are always using the latest schema for every org

Reply:
Makes sense, thanks!